### PR TITLE
(PA-715) Add Ubuntu 16.10 platform (i386, x86_64)

### PR DESCRIPTION
**Pull request by ScottGarman**
Builds correctly. Will update with runtime smoke testing shortly.

### CODE REVIEW

**Comment by ScottGarman** [comment]
Smoke testing passes - puppet --version works, and I can apply a simple manifest that uses the file resource to create a file. facter os shows:

```
{
  architecture => "i386",
  distro => {
    codename => "yakkety",
    description => "Ubuntu 16.10",
    id => "Ubuntu",
    release => {
      full => "16.10",
      major => "16.10"
    }
  },
  family => "Debian",
  hardware => "i686",
  name => "Ubuntu",
  release => {
    full => "16.10",
    major => "16.10"
  },
  selinux => {
    enabled => false
  }
}
```

File: configs/platforms/ubuntu-16.10-amd64.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  plat.servicedir "/lib/systemd/system"
+  plat.defaultdir "/etc/default"
+  plat.servicetype "systemd"
+  plat.codename "yakkety"
+
+  plat.add_build_repository "http://pl-build-tools.delivery.puppetlabs.net/debian/pl-build-tools-release-#{plat.get_codename}.deb"
+  plat.provision_with "export DEBIAN_FRONTEND=noninteractive; apt-get update -qq; apt-get install -qy --no-install-recommends build-essential devscripts make quilt pkg-config debhelper rsync fakeroot"
+  plat.install_build_dependencies_with "DEBIAN_FRONTEND=noninteractive; apt-get install -qy --no-install-recommends "
+  plat.vmpooler_template "ubuntu-1610-x86_64"
+  plat.output_dir File.join("deb", plat.get_codename, "PC1")
```
Comment (ScottGarman) [inline configs/platforms/ubuntu-16.10-amd64.rb]: I'm not entirely sure about this inconsistency either, but the ubuntu platforms seem to all add it, while the feodora ones to do. @McdonaldSeanp may have more backstory on that.

Comment (ScottGarman) [inline configs/platforms/ubuntu-16.10-amd64.rb]: From the looks of things, the output directory for my fedora builds has PC1 in it already:

```
output/
output/fedora
output/fedora/f25
output/fedora/f25/PC1
output/fedora/f25/PC1/x86_64
output/fedora/f25/PC1/x86_64/puppet-agent-1.8.2.346.g8c0eb4a-1.fedoraf25.x86_64.rpm
output/fedora/f25/PC1/i386
output/fedora/f25/PC1/i386/puppet-agent-1.8.2.345.g75b49a3-1.fedoraf25.i386.rpm
```

**Comment by Magisus** [comment]
Looks good to me 👍

**Pull request merged**